File: 919a.py
# maybe better built in accuracy than cpp
[n,m] = [int(x) for x in input().split()]
smallNum = -1
for i in range(n):
    [t,v] = [int(x) for x in input().split()]
    if smallNum == -1:
        smallNum = m*(t/v)
    else:
        smallNum = min(smallNum, m*(t/v))
print(smallNum)

File: 919a.cpp
#include <iostream>

using namespace std;

int main()  {
    int n,m;
    double smallNum = -1;
    cin >> n >> m;
    
    for (int i = 0; i < n; ++i) {
        double t,v;
        cin >> t >> v;
        smallNum = min(smallNum, m*(t/v));
        if (smallNum == -1) smallNum = m*(t/v);
    }
    cout << smallNum;
    return 0;
}
